lib/Dialect/FIRRTL/Import/FIRLexer.cpp

```diff
@@ -148,28 +148,29 @@ std::string FIRToken::getRawStringValue(StringRef spelling) {
 //===----------------------------------------------------------------------===//
 
 static StringAttr getMainBufferNameIdentifier(const llvm::SourceMgr &sourceMgr,
-                                              MLIRContext *context) {
-  auto mainBuffer = sourceMgr.getMemoryBuffer(sourceMgr.getMainFileID());
+                                              MLIRContext *context,
+                                              unsigned bufferID) {
+  auto mainBuffer = sourceMgr.getMemoryBuffer(bufferID);
   StringRef bufferName = mainBuffer->getBufferIdentifier();
   if (bufferName.empty())
     bufferName = "<unknown>";
   return StringAttr::get(context, bufferName);
 }
```

Comment:
Can you update the name of this function since it doesn't have to be used for the main buffer anymore. Maybe `getBufferIdentifier`? Also`s/mainBuffer/buffer`

Reply:
Done and done. I kept the name structure as `getBufferNameIdentifier` since it's still returning the name and that's what the class member is called. I could change it to `getBufferNameStringAttr` or something but it already specifies that it's a StringAttr in the return type.